modules/javafx.controls/src/main/java/javafx/scene/control/skin/LabeledSkinBase.java

```diff
@@ -324,26 +325,27 @@ protected void updateChildren() {
             widthPadding += leftLabelPadding() + rightLabelPadding();
         }
 
-        double textWidth = 0.0;
-        if (!emptyText) {
-            textWidth = Utils.computeTextWidth(font, cleanText, 0);
-        }
-
-        // Fix for RT-39889
-        double graphicWidth = graphic == null ? 0.0 :
-                Utils.boundedSize(graphic.prefWidth(-1), graphic.minWidth(-1), graphic.maxWidth(-1));
+        double textWidth = emptyText ? 0.0 : Utils.computeTextWidth(font, cleanText, 0);
```

Comment:
textWidth is being computed even if isIgnoreText() is true.  I wonder if we could re-structure the if-else statements below to avoid unnecessary computations?

There is another suggestion - isIgnoreText() is not a simple getter (does a bit of computation).  We probably should use a boolean early in this method to avoid calling it repeatedly.

Reply:
Previously both `textWidth` and `graphicWidth` were calculated regardless of the condition. In the updated code calculation of `graphicWidth` is moved to `else` block so that one unnecessary calculation is avoided. Since `textWidth` is used in 2 more conditions in `else` block, calculated it before the start of `if` block.

Since `isIgnoreText()` does more than just checking for `ContentDisplay` value, further reordering `if-else` statements generates different result than expected.
So this further leads to question previously raised by @hjohn, if `isIgnoreText()` and `isIgnoreGraphic()` should only check for ContetDisplay type or check for conditions like null/empty graphic or text as well. If this needs to be separated, again it gives raise to more conditions in the `computePrefWidth()` method.

Added boolean in the beginning as suggested.

Reply:
Updated `computePrefWidth()` method.
Please refer to my comment at the end regarding `computePrefHeight()` method optimization observations.